### ISSUE TITLE
react-a11y-anchors should accept content when given a child component

### ISSUE THREAD

**Issue by astorije** [Microsoft/tslint-microsoft-contrib#433]
`react-a11y-anchors` passes on the following snippets (assuming each line is a different file as I am focusing this issue on text and not `href`):

```tsx
<a href="https://example.com">Hello world</a>
<a href="https://example.com"><TextWrapper>Hello world</TextWrapper></a>
```

However, it fails on the following ones:

```tsx
<a href="https://example.com"><TextWrapper /></a>
<a href="https://example.com"><TextWrapper value="Hello world" /></a>
```

It's not an uncommon situation, for example when rendering translated strings.

I think it would be reasonable to take [`eslint-plugin-jsx-a11y`'s approach](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md): accept any React component as valid content, unless it has a `aria-hidden` prop.
Thoughts?

**Comment by br1anchen** [comment]
@astorije According to [eslint-plugin-jsx-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md), it should pass:
```
<a dangerouslySetInnerHTML={{ __html: 'foo' }} />
```
I am not sure if we should support this too, it might be out of this issue scope here.
